ui/controlgroup.js

```diff
@@ -0,0 +1,175 @@
+			"checkboxradio": "input[type='checkbox'], input[type='radio']",
+			"selectmenu": "select"
+		},
+		direction: "horizontal",
+		excludeInvisible: true,
+		classes: {}
+	},
+
+	_create: function() {
+		this._enhance();
+	},
+
+	// To support enhanced option in jQuery mobile we isolate dom manipulation here
+	_enhance: function() {
+		this.element.attr( "role", "toolbar" );
+		this.refresh();
+	},
+
+	_destroy: function() {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			that.element.children( selector ).map(function(){
+				return $( this )[ widget ]( "widget" )[ 0 ];
+			}).removeData( "ui-controlgroup-data" );
+		});
+		this._callChildMethod( "destroy" );
+		this.element.removeAttr( "role" );
+	},
+
+	_callChildMethod: function( method ) {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			var options = {};
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+			if ( $.fn[ widget ] && selector ) {
+				that.element
+					.children( selector )[ widget ]( method ? method: options )
+					.each(function(){
+						if ( method !== "destroy" ) {
+							$( this )[ widget ]( "widget" ).data( "ui-controlgroup-data", {
+								"type": widget,
+								"element": $( this )
+							});
+						}
+					});
+			}
+		});
+	},
+
+	_button_options: function( position ) {
+		var cornerClasses = {
+			"middle": null,
+			"first": "ui-corner-" + ( ( this.options.direction === "vertical" )? "top" : "left" ),
+			"last": "ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom" : "right" )
+		};
+
+		return {
+					classes: {
+						"ui-button": cornerClasses[ position ]
+					}
+				};
+	},
+
+	_checkboxradio_options: function( position ) {
+		var cornerClasses = {
+			"middle": null,
+			"first": "ui-corner-" + ( ( this.options.direction === "vertical" )? "top" : "left" ),
+			"last": "ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom" : "right" )
+		};
+
+		return {
+					classes: {
+						"ui-checkboxradio-label": cornerClasses[ position ]
+					}
+				};
+	},
+
+	_selectmenu_options: function( position ) {
+		var classes = {
+			middle: {
+				"ui-selectmenu-button-open": null,
+				"ui-selectmenu-button-closed": null
+			},
+			first: {
+				"ui-selectmenu-button-open":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "top": "tl" ),
+				"ui-selectmenu-button-closed":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "top": "left" )
+			},
+			last: {
+				"ui-selectmenu-button-open":
+					( this.options.direction === "vertical" )? null: "ui-corner-tr",
+				"ui-selectmenu-button-closed":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom": "right" )
+			}
+
+		};
+		return { classes: classes[ position ] };
+	},
+
+	_setOption: function( key, value ) {
+		var original = this.options[ key ];
```

Comment:
This variable is used just once, could inline it by changing the order of the calls below.